File: Tutorial1-1DaggerBasics/src/main/java/com/example/tutorial1daggerbasics/MyApplication.java
package com.example.tutorial1daggerbasics;

import android.app.Application;

import com.example.tutorial1daggerbasics.di.ApplicationComponent;
import com.example.tutorial1daggerbasics.di.ApplicationModule;
import com.example.tutorial1daggerbasics.di.DaggerApplicationComponent;

public class MyApplication extends Application {
    private ApplicationComponent mMyComponent;

    @Override
    public void onCreate() {
        super.onCreate();
        mMyComponent = createMyComponent();
    }

    ApplicationComponent getMyComponent() {
        return mMyComponent;
    }

    private ApplicationComponent createMyComponent() {
        return DaggerApplicationComponent.builder()
                .applicationModule(new ApplicationModule())
                .build();
    }
}

File: Tutorial1-1DaggerBasics/src/main/java/com/example/tutorial1daggerbasics/MyApplication.kt
package com.example.tutorial1daggerbasics

import android.app.Application
import com.example.tutorial1daggerbasics.di.ApplicationComponent
import com.example.tutorial1daggerbasics.di.ApplicationModule
import com.example.tutorial1daggerbasics.di.DaggerApplicationComponent

class MyApplication : Application() {

    lateinit var applicationComponent: ApplicationComponent
        private set

    override fun onCreate() {
        super.onCreate()
        applicationComponent = createMyComponent()
    }

    private fun createMyComponent(): ApplicationComponent {
        return DaggerApplicationComponent.builder()
                .applicationModule(ApplicationModule())
                .build()
    }

}